From 15d743662b29465641e1c951bb2889abd1d560b4 Mon Sep 17 00:00:00 2001
From: Tim Orling <tim.orling@konsulko.com>
Date: Mon, 25 Jul 2022 19:24:07 -0700
Subject: [PATCH] libchrome: base/time/time_delta_from_string cast

Fix errors like:

error: implicit conversion from 'int64_t' (aka 'long') to 'double' may
lose precision [-Werror,-Wimplicit-int-float-conversion]
|           (static_cast<double>(sign) * number.frac_part / number.frac_scale) *
|                                                         ~ ~~~~~~~^~~~~~~~~~
|
error: implicit conversion from 'int64_t' (aka 'long') to 'double' may
lose precision [-Werror,-Wimplicit-int-float-conversion]
|           (static_cast<double>(sign) * number.frac_part / number.frac_scale) *
|                                      ~ ~~~~~~~^~~~~~~~~
Signed-off-by: Tim Orling <tim.orling@konsulko.com>
Change-Id: Ifb4794bd856bab61a9c95116598fa44dca8cf8fe
---
 base/time/time_delta_from_string.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/base/time/time_delta_from_string.cc
===================================================================
--- a/base/time/time_delta_from_string.cc
+++ b/base/time/time_delta_from_string.cc
@@ -150,7 +150,7 @@ absl::optional<TimeDelta> TimeDeltaFromS
       delta += sign * number.int_part * unit;
     if (number.frac_part != 0)
       delta +=
-          (static_cast<double>(sign) * number.frac_part / number.frac_scale) *
+          (static_cast<double>(sign) * static_cast<double>(number.frac_part / number.frac_scale)) *
           unit;
   }
   return delta;
